lib/src/core/engine.dart

```diff
@@ -709,3 +742,59 @@ extension EngineInternalMethods on Engine {
         _lossyDCPub
       ].whereNotNull().map((e) => e.toLKInfoType()).toList();
 }
+
+Future<String?> getConnectedAddress(rtc.RTCPeerConnection pc) async {
+  var selectedCandidatePairId = '';
+  final candidatePairs = <String, rtc.StatsReport>{};
+  // id -> candidate ip
+  final candidates = <String, String>{};
+  final List<rtc.StatsReport> stats = await pc.getStats();
+  for (var v in stats) {
+    switch (v.type) {
+      case 'transport':
+      case 'googComponent':
+        selectedCandidatePairId = v.values['selectedCandidatePairId'] as String;
+        break;
+      case 'candidate-pair':
```

Comment:
how come they are still coming through via these different variations? Isn't the same libwebrtc that it's building against?

Reply:
I haven't migrated to the standard API of getStats yet, because the Android SDK misses getStats for RtpSender/RtpReceiver, and needs a small patch for libwebrtc Android.

Reply:
I see, seems a bit messy to propagate this to clients of the library, but good to know how to clean this up in the future.